src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11HKDF.java

```diff
@@ -216,6 +219,10 @@ private <T> T derive(String alg, AlgorithmParameterSpec derivationSpec,
             }
             return retType.cast(ret);
         } catch (PKCS11Exception e) {
+            if (e.match(CKR_KEY_SIZE_RANGE)) {
+                throw new InvalidAlgorithmParameterException("Invalid key " +
+                        "size for algorithm '" + alg + "'.", e);
```

Comment:
nit: include the requested key size in the exception message?

Reply:
Sounds good!